crates/bevy_animation/src/lib.rs

```diff
@@ -691,217 +617,269 @@ where
-    children: &Query<&Children>,
+    animation_clips: &Assets<AnimationClip>,
 ) {
-    let Some(animation_clip) = animations.get(&animation.animation_clip) else {
+    let Some(animation_clip) = animation_clips.get(&animation.animation_clip) else {
         return;
     };
 
-    // We don't return early because seek_to() may have been called on the animation player.
-    animation.update(
-        if paused { 0.0 } else { time.delta_seconds() },
-        animation_clip.duration,
-    );
+    let delta = if paused { 0.0 } else { time.delta_seconds() };
 
-    if animation.path_cache.len() != animation_clip.paths.len() {
-        let new_len = animation_clip.paths.len();
-        animation.path_cache.iter_mut().for_each(|v| v.clear());
-        animation.path_cache.resize_with(new_len, Vec::new);
-    }
-    if !verify_no_ancestor_player(maybe_parent, parents) {
-        warn!("Animation player on {:?} has a conflicting animation player on an ancestor. Cannot safely animate.", root);
-        return;
-    }
+    animation.update(delta, animation_clip.duration);
+}
 
-    let mut any_path_found = false;
-    for (path, bone_id) in &animation_clip.paths {
-        let cached_path = &mut animation.path_cache[*bone_id];
-        let curves = animation_clip.get_curves(*bone_id).unwrap();
-        let Some(target) = entity_from_path(root, path, children, names, cached_path) else {
-            continue;
+impl PlayingAnimation {
+    fn apply(&self, clips: &Assets<AnimationClip>, weight: f32, target_entity: &mut EntityMut) {
+        let target = target_entity.get::<AnimationTarget>().unwrap();
+
+        let Some(clip) = clips.get(&self.animation_clip) else {
+            warn!(
+                "Couldn't find the animation clip for {:?}",
+                target_entity.id()
+            );
+            return;
         };
-        any_path_found = true;
-        // SAFETY: The verify_no_ancestor_player check above ensures that two animation players cannot alias
-        // any of their descendant Transforms.
-        //
-        // The system scheduler prevents any other system from mutating Transforms at the same time,
-        // so the only way this fetch can alias is if two AnimationPlayers are targeting the same bone.
-        // This can only happen if there are two or more AnimationPlayers are ancestors to the same
-        // entities. By verifying that there is no other AnimationPlayer in the ancestors of a
-        // running AnimationPlayer before animating any entity, this fetch cannot alias.
-        //
-        // This means only the AnimationPlayers closest to the root of the hierarchy will be able
-        // to run their animation. Any players in the children or descendants will log a warning
-        // and do nothing.
-        let Ok(mut transform) = (unsafe { transforms.get_unchecked(target) }) else {
-            continue;
+
+        let Some(curves) = clip.get_curves(target.id) else {
+            return;
```

Comment:
Do we wan't to warn here too? I can't think of many reasons this would happen other than misconfiguration.

Reply:
It actually fires a lot during `animated_fox`, as bones that aren't being animated by the current clip won't have a curve.

Reply:
Ah, of-course. Considered resolved.